cmake/UnitTests.cmake

```diff
@@ -2358,6 +2357,31 @@ list(APPEND CPU_POCL_FAILED_TESTS "Unit_deviceFunctions_CompileTest_y1f_float")
 list(APPEND CPU_POCL_FAILED_TESTS "Unit_deviceFunctions_CompileTest_ynf_float") # Failed
 list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipClassKernel_Friend") # SEGFAULT
 list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipDeviceSynchronize_Functional") # Failed
+# broken tests, they all try to write outside allocated memory;
+# valgrind + pocl shows:
+#
+#==11492== Invalid write of size 1
+#==11492==    at 0x5605B83: pocl_fill_aligned_buf_with_pattern (pocl_util.c:2590)
+#==11492==    by 0x562B1F5: pocl_driver_svm_fill (common_driver.c:444)
+#==11492==    by 0x5626EBF: pocl_exec_command (common.c:693)
+#==11492==    by 0x562205F: pthread_scheduler_get_work (pthread_scheduler.c:529)
+#==11492==    by 0x56221B7: pocl_pthread_driver_thread (pthread_scheduler.c:588)
+#==11492==    by 0x5009B42: start_thread (pthread_create.c:442)
+#==11492==    by 0x509ABB3: clone (clone.S:100)
+#==11492==  Address 0x114baf00 is 0 bytes after a block of size 16,384 alloc'd
+#==11492==    by 0x561EE67: pocl_basic_svm_alloc (basic.c:841)
+#==11492==    by 0x561411A: POclSVMAlloc (clSVMAlloc.c:98)
+#==11492==    by 0x4966168: SVMemoryRegion::allocate(unsigned long) (source/chip-spv/src/backend>
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2D_BasicFunctional")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2DAsync_BasicFunctional")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2DAsync_WithKernel")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemset2DAsync_MultiThread")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetFunctional_ZeroValue_2D")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetASyncMulti")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetDASyncMulti - int8_t")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetDASyncMulti - int16_t")
+list(APPEND CPU_POCL_FAILED_TESTS "Unit_hipMemsetDASyncMulti - uint32_t")
```

Comment:
Can you add how these tests fail so that we're in line with the rest of the tests?

`# Failed/SEGFAULT/Subprocess Aborted`

Reply:
Added the failures.

I've added some [extra SVM pointer checks](https://github.com/pocl/pocl/pull/1171) in PoCL. With those, the first five tests (Memset2D) now "pass" - they get a `CL_INVALID_OPERATION` from PoCL and.... AFAICT they simply ignore it. So it's a "pass" but really a fail. BTW same thing happens with Intel OpenCL CPU runtime.

Reply:
> So it's a "pass" but really a fail.

are you saying that the tests don't check the result of the HIP API call? Still don't understand why these tests are not failing on the CI system.

Reply:
> are you saying that the tests don't check the result of the HIP API call

That's my understanding. Here's a log from `ctest -V -R Unit_hipMemsetFunctional_ZeroValue_2D`:

```
test 860
    Start 860: Unit_hipMemsetFunctional_ZeroValue_2D

860: Test command: /home/michal/0/build/b_chip_15/catch/catch_tests/unit/memory/hipMemsetFunctional "Unit_hipMemsetFunctional_ZeroValue_2D"
860: Test timeout computed to be: 10000000
860: CHIP error [TID 19237] [1675784262.716106613] : hipErrorRuntimeMemory (CL_INVALID_VALUE ) in /home/michal/0/source/chip-spv/src/backend/OpenCL/CHIPBackendOpenCL.cc:1048:memFillAsyncImpl
860: 
860: CHIP error [TID 19237] [1675784262.716161386] : Caught Error: hipErrorRuntimeMemory
860: CHIP error [TID 19237] [1675784262.732102895] : hipErrorRuntimeMemory (CL_INVALID_VALUE ) in /home/michal/0/source/chip-spv/src/backend/OpenCL/CHIPBackendOpenCL.cc:1048:memFillAsyncImpl
860: 
860: CHIP error [TID 19237] [1675784262.732118556] : Caught Error: hipErrorRuntimeMemory
860: Filters: Unit_hipMemsetFunctional_ZeroValue_2D
860: ===============================================================================
860: All tests passed (32768 assertions in 1 test case)
860: 
1/1 Test #860: Unit_hipMemsetFunctional_ZeroValue_2D ...   Passed    0.62 sec
```

Reply:
What is the issue with the log that you attached? @franz

Reply:
the issue with the log is that the test gets CL_INVALID_VALUE from one of the OpenCL calls, but the overall result (reported by ctest) is still "Passed". So either the test itself doesn't check the HIP call result, or there is a bug in CHIP-SPV and the result is ignored somewhere.

Reply:
Maybe the tests are checking fot invalid parameters so an error must be thrown in order to pass the test.

Reply:
Possible but unlikely. With the OpenCL intel GPU as backend, the test doesn't print any HIP error and still passes.

Reply:
I checked at least this one case and it's passing nullptr for device pointer so this is likely an argument check issue.